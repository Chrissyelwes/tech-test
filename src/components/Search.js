// eslint-disable-next-line
import React, { useState } from "react";
import "../styles/search.css";
import getImages from "../requests/getImages";

const Search = () => {
    // eslint-disable-next-line
    const [value, setValue] = useState();

    const handleSubmit = (event) => {
        event.preventDefault();
        getImages(value);
    };

    return(
        <div className="search">
        <form className="search-form" onSubmit={handleSubmit}>
        <input
            className="search-input"
            type="text" 
            onChange={(e) => setValue(e.target.value)}
        />
        {/* this will save the users search query every time there is a change within the input */}
            <button className="search-btn" type="submit">
                Search
            </button>
        </form>
        </div>
    )
};

export default Search;
