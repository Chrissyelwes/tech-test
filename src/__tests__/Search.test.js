import React from "react";
import { render, screen } from '@testing-library/react';
import Search from '../components/Search';

describe("Search", () => {
    const { asFragment } = render(<Search />)

    test("renders Search component correctly", () => {
        expect(asFragment()).toMatchSnapshot();
    });
});