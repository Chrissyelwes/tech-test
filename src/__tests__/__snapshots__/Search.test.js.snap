// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders Search component correctly 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <form
      class="search-form"
    >
      <input
        class="search-input"
        type="text"
      />
      <button
        class="search-btn"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</DocumentFragment>
`;
